boardswarm-cli/src/ui.rs

```diff
@@ -246,6 +350,18 @@ pub async fn run_ui(
             .await
     });
 
+    let _device_state_watcher = tokio::spawn(async move {
+        let mut d = dev_clone.device_info().await.unwrap();
+        let mut current_mode = None;
```

Comment:
The device mode already knows the current mode as it will wait for a first info on creation.